tests/test_admin.py

```diff
@@ -647,3 +647,127 @@ def test_cms_page_toolbar_pages_link_doesnt_filter_the_page_list_by_page_id(self
+            "expiry_date": 1,
+            "version_state": 2,
+            "version_author": 3,
+            "url": 4,
+            "compliance_number": 5,
+        }
+        self.export_admin_endpoint = self.get_admin_url(PageContent, "export_csv")
+        self.template_1 = get_cms_setting('TEMPLATES')[0][0]
+
+    def test_export_button_endpoint_response_is_a_csv(self):
+        """
+        Valid csv file is returned from the admin export endpoint
+        """
+        PageContentWithVersionFactory.create_batch(6, template=self.template_1, language="en")
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(self.export_admin_endpoint)
+
+        # Endpoint is returning 200 status code
+        self.assertEqual(response.status_code, 200)
+        # Response contains a csv file
+        self.assertEquals(
+            response.get('Content-Disposition'),
+            "attachment; filename={}.csv".format("cms.pagecontent")
+        )
+
+    def test_export_content_headers(self):
+        """
+        Export should contain all the headings in the current page content list display
+        """
+        PageContentWithVersionFactory()
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(self.export_admin_endpoint)
+
+        csv_headings = response.content.decode().splitlines()[0].split(",")
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(
+            csv_headings[self.headings_map["title"]],
+            "Title"
+        )
+        self.assertEqual(
+            csv_headings[self.headings_map["expiry_date"]],
+            "Expiry Date"
+        )
+        self.assertEqual(
+            csv_headings[self.headings_map["version_state"]],
+            "Version State"
+        )
+        self.assertEqual(
+            csv_headings[self.headings_map["version_author"]],
+            "Version Author"
+        )
+        self.assertEqual(
+            csv_headings[self.headings_map["url"]],
+            "Url"
+        )
+        self.assertEqual(
+            csv_headings[self.headings_map["compliance_number"]],
+            "Compliance Number"
+        )
+
+    def test_file_content_contains_values(self):
+        """
+        CSV response should contain expected values.
+        """
+        version = PageVersionFactory(state=PUBLISHED, content__language="en")
+        preview_url = get_object_preview_url(version)
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(self.export_admin_endpoint)
+
+        self.assertEqual(response.status_code, 200)
+
+        csv_lines = response.content.decode().splitlines()
+
+        content_row_1 = csv_lines[1].split(",")
+
+        self.assertIn(
+            content_row_1[self.headings_map["title"]],
+            version.content.title
+        )
+        self.assertEqual(
+            content_row_1[self.headings_map["expiry_date"]],
```

Comment:
I tried to avoid making djangocms-pageadmin dependent on content-expiry. I will have a think about this one

Reply:
I hadn't considered that, but see what you mean. I believe that `djangocms_alias` has some tests that enable/disable versioning for this reason, although it might be a bit overkill to replicate that for just this, as it is used more extensively there. I'm not sure how familiar you are with using mocks? But that could be an alternative, particularly if used to unit tests the `get_compliance_number` and `get_expiry_date` methods directly. An example below:
```
    def test_get_compliance_number(self):
        mock_with_content_expiry = MagicMock(spec=Version)
        mock_with_content_expiry.contentexpiry = MagicMock(compliance_number="123456")

        page_content = PageContent()
        model_admin = PageContentAdmin(PageContent, admin.AdminSite())
        with patch.object(model_admin, "get_version") as mock_get_version:
            mock_get_version.return_value = mock_with_content_expiry
            result = model_admin.get_compliance_number(page_content)

            self.assertEqual(result, "123456")
            mock_get_version.assert_called_once_with(page_content)
```

Reply:
Cool, thanks for this. Added tests for both methods